More misc changes, mostly editorial (#1155)
moq-wg/moq-transport: draft-ietf-moq-transport.md

* Start Location : The starting location.

> * End Group: The end Group ID. A value of 0 means the subscription is
> open-ended.

Reviewer: ```suggestion
* End Group: The end Group ID, plus 1. A value of 0 means the subscription is
open-ended.
```

|------|------------|

The LSB of the type determines if the Extensions Headers Length and Extension

> headers are present. If an endpoint receives a datagram with Type 0x21 and
> Extension Headers Length is 0, it MUST close the session with PROTOCOL_VIOLATION.

Reviewer: ```suggestion
headers are present. If an endpoint receives a datagram with Extensions Present
and Extension Headers Length is 0, it MUST close the session with
`PROTOCOL_VIOLATION`.
```

## Streams

> When objects are sent on streams, the stream begins with a Subgroup or Fetch
> Header and is followed by one or more sets of serialized object fields.
> If a stream ends gracefully (i.e., the stream terminates with a FIN) in the
> middle of a serialized Object, the session SHOULD be terminated with a
> PROTOCOL_VIOLATION.

Reviewer: ```suggestion
When Objects are sent on streams, the stream begins with a Subgroup or Fetch
Header and is followed by one or more sets of serialized Object fields.
If a stream ends gracefully (i.e., the stream terminates with a FIN) in the
middle of a serialized Object, the session SHOULD be terminated with a
`PROTOCOL_VIOLATION`.
```

Processing a RESET_STREAM or RESET_STREAM_AT means that there might be other
objects in the Subgroup beyond the last one received. A relay might immediately
reset the corresponding downstream stream, or it might attempt to recover the

> missing Objects in an effort to send all the objects in the subgroups and the FIN.
> It also might send RESET_STREAM_AT with reliable_size set to the last object it
> has, so as to reliably deliver the objects it has while signaling that other
> objects might exist.

Reviewer: ```suggestion
missing Objects in an effort to send all the Objects in the subgroups and the FIN.
It also might send RESET_STREAM_AT with reliable_size set to the last Object it
has, so as to reliably deliver the Objects it has while signaling that other
Objects might exist.
```
